models/release.go

```diff
@@ -357,6 +359,17 @@ func DeleteReleaseByID(id int64, doer *User, delTag bool) error {
 		return fmt.Errorf("LoadAttributes: %v", err)
 	}
 
+	if _, err := x.Delete(&Attachment{ReleaseID: id}); err != nil {
```

Comment:
imho this should be moved below after files have been deleted

Reply:
https://github.com/go-gitea/gitea/pull/6068#discussion_r325625942

@lunny requested that. If there’s a consensus, will do tgat

Reply:
@lafriks I don't think so. If you deleted files successfully but database record failed. then it will break. But if you deleted database record successfully but files failed, it only has some garbages. But it works.

Reply:
Still it's tricky but ok :)